mpf/devices/blinkenlight.py

```diff
@@ -0,0 +1,131 @@
+"""Contains the Blinkenlight class."""
+import asyncio
+from functools import partial
+
+from typing import Set, Dict, List, Tuple, Any
+
+from mpf.core.rgb_color import RGBColor, ColorException
+from mpf.core.system_wide_device import SystemWideDevice
+from mpf.core.delays import DelayManager
+from mpf.core.device_monitor import DeviceMonitor
+from mpf.exceptions.config_file_error import ConfigFileError
+
+@DeviceMonitor("num_colors", "light")
+class Blinkenlight(SystemWideDevice):
+
+    """A light that alternates between several different colors."""
+
+    config_section = 'blinkenlights'
+    collection = 'blinkenlights'
+    class_label = 'blinkenlight'
+
+    def __init__(self, machine, name):
+        """Initialise blinkenlight."""
+        super().__init__(machine, name)
+
+        self._colors = []
+        self.delay = DelayManager(machine)
+        self._num_colors = 0
+
+    def load_config(self, config: dict):
+        super().load_config(config)
+
+        self._color_duration = self.config['color_duration']
+        self._cycle_duration = self.config['cycle_duration']
+        if (self._color_duration is None and self._cycle_duration is None) or (self._color_duration is not None and self._cycle_duration is not None):
+            self._blinkenlight_validation_error("Either color_duration or cycle_duration must be specified, but not both.", 1)
+
+    def _blinkenlight_validation_error(self, msg, error_code) -> "NoReturn":  # pragma: no cover
+        raise ConfigFileError('"{}" >> {}'.format(self.name, msg), error_code, "blinkenlight", self.name)
+
+    @property
+    def num_colors(self):
+        # I would prefer this to be a calculated value (i.e., just return len(self._colors)), but then there's no setter involved so things like mpf monitor won't know when this value gets updated.
```

Comment:
There are some magic methods on DeviceMonitor to notify it about changes.

Reply:
I tried to find the magic methods you mentioned but I don't see any on DeviceMonitor aside from `__init__` and `__call__`.  I know `__setattr__` is used to notify subscribers when a variable is changed, but how can I notifiy subscribers without creating a fake variable like this?